modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/FormsRuleEditor.es.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {cleanup, render} from '@testing-library/react';
+import Select from 'dynamic-data-mapping-form-field-type/Select/Select.es';
+import React from 'react';
+
+import {FormsRuleEditor} from '../../../../../src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/FormsRuleEditor.es';
+import {DEFAULT_RULE} from '../../../../../src/main/resources/META-INF/resources/data_layout_builder/js/components/rule-builder/editor/config.es';
+import {FIELDS, OPERATORS_BY_TYPE} from '../../../../mock/fields.es';
+
+global.fetch.enableFetchMocks();
+
+const pages = [
+	{
+		label: '1 Page title',
+		name: '0',
+		value: '0',
+	},
+	{
+		label: '2 Page title',
+		name: '1',
+		value: '1',
+	},
+	{
+		label: '3 Page title',
+		name: '2',
+		value: '2',
+	},
+];
+
+const rule = {
+	actions: [
+		{
+			action: 'show',
+			label: 'radio',
+			target: 'radio',
+		},
+	],
+	conditions: [
+		{
+			operands: [
+				{
+					label: 'text',
+					repeatable: false,
+					source: 0,
+					type: 'field',
+					value: 'text',
+				},
+				{
+					type: 'string',
+					value: 'aaaa',
+				},
+			],
+			operator: 'equals-to',
+		},
+	],
+	'logical-operator': 'OR',
+};
+
+describe('FormsRuleEditor', () => {
+	const originalLiferayLoader = window.Liferay.Loader;
+	afterEach(() => {
+		cleanup();
+		jest.restoreAllMocks();
+	});
+
+	beforeEach(() => {
+		global.fetch.mockResponse(
+			JSON.stringify([
+				{
+					javaScriptModule: 'select',
+					name: 'select',
+				},
+			])
+		);
+	});
+
+	beforeAll(() => {
+		window.Liferay = {
+			...window.Liferay,
+			Loader: {
+				require: ([fieldModule], resolve) => {
+					switch (fieldModule) {
+						case 'select':
+							resolve({default: Select});
+							break;
+						default:
+							break;
+					}
+				},
+			},
+		};
+	});
+
+	afterAll(() => {
+		window.Liferay.Loader = originalLiferayLoader;
+	});
```

Comment:
Could We place this mock on module's `jest.config.js`?

/cc @matuzalemsteles Does it make sense?

Reply:
@diegonvs I think it could but this implementation is very exclusive to this test suite, `Liferay.Loader.require` can load any file, it would be strange to just load the fields as is the case here.